### PR TITLE
Project Beats: trigger events don't update during playback if modified

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
This fixes a small bug where if code under a trigger block was modified during playback, after being triggered, changes wouldn't not be reflected in the song and the timeline. The reason for this was that on block space change, we were only re-executing code under the when run block. To fix this, we keep track of all triggers that have been triggered during playback, and on block space change, we re-execute their code in addition to the when_run code. To make sure they play at the original time that they were scheduled, we pass a `startPosition` arg to the function call which the generated blockly code uses, instead of querying the player for the current playhead time directly. This way, when the trigger is first triggered, we can pass the current playhead time, but if it needs to be re-executed after code has changed, we have a way to pass the original time at which it was triggered.

A byproduct of this is that we don't need to separately manage and clear when_run and triggered events, because all events that need to get rescheduled when the workspace updates will get queued automatically. There is a separate discussion on whether we'd like to remove the distinction between triggered and when_run events in our models entirely, but there may be benefits to keeping that information around, for example if we'd like to render triggered events differently.

## Links

https://codedotorg.atlassian.net/browse/SL-748

## Testing story

Tested locally on all block models.

### CODE REVIEW

File: apps/src/music/player/MusicPlayer.ts [hunk 1]
```diff
@@ -297,7 +281,7 @@ export default class MusicPlayer {
   // Converts actual seconds used by the audio system into a playhead
   // position, which is 1-based and scaled to measures.
   convertSecondsToPlayheadPosition(seconds: number): number {
-    return 1 + seconds / this.secondsPerMeasure();
+    return Math.round((1 + seconds / this.secondsPerMeasure()) * 1000) / 1000;
```
Comment (breville) [inline apps/src/music/player/MusicPlayer.ts]: Curious: why this change?

**Review by breville — APPROVED** [review]
Tested locally and this works nicely!  Good catch and fix.

File: apps/src/music/player/MusicPlayer.ts (hunk 1 above)
Comment (sanchitmalhotra126) [reply apps/src/music/player/MusicPlayer.ts]: ah good catch, was leftover from a different approach where I was using the playhead time as an event identifier. though, maybe it's worth keeping this around if we want to keep playhead values rounded? I was seeing values like 1.9999999998

Comment (sanchitmalhotra126) [reply apps/src/music/player/MusicPlayer.ts]: Discussed offline - removing rounding to preserve precision.
